srf_generation/input_file_generation/realisation_to_srf.py

```diff
@@ -0,0 +1,292 @@
+    elif target_slip_cm is not None:
+        dd = np.sqrt(moment * 1.0e-20) / (target_slip_cm * vs * vs * rho)
+        slip = target_slip_cm
+    else:
+        aa = np.exp(2.0 / 3.0 * np.log(moment) - 14.7 * np.log(10.0))
+        dd = np.sqrt(aa)
+        slip = (moment * 1.0e-20) / (aa * vs * vs * rho)
+
+    ###
+    ### file names
+    ###
+    srf_file = simulation_structure.get_srf_path(cs_root, name)
+    gsf_file = srf_file.replace(".srf", ".gsf")
+    makedirs(path.dirname(srf_file), exist_ok=True)
+
+    ###
+    ### create GSF
+    ###
+    with open(gsf_file, "w") as gsfp:
+        gsfp.write("# nstk= 1 ndip= 1\n")
+        gsfp.write(f"# flen= {dd:10.4f} fwid= {dd:10.4f}\n")
+        gsfp.write(
+            "# LON  LAT  DEP(km)  SUB_DX  SUB_DY  LOC_STK  LOC_DIP  LOC_RAKE  SLIP(cm)  INIT_TIME  SEG_NO\n"
+        )
+        gsfp.write("1\n")
+        gsfp.write(
+            f"{longitude:11.5f} {latitude:11.5f} {depth:8.4f} {dd:8.4f} {dd:8.4f} "
+            f"{strike:6.1f} {dip:6.1f} {rake:6.1f} {slip:8.2} {inittime:8.3f}    0\n"
+        )
+
+    ###
+    ### create SRF
+    ###
+    commands = [
+        binary_version.get_unversioned_bin(GENERICSLIP2SRF),
+        f"infile={gsf_file}",
+        f"outfile={srf_file}",
+        "outbin=0",
+        f"stype={stype}",
+        f"dt={dt}",
+        "plane_header=1",
+        f"risetime={risetime}",
+        "risetimefac=1.0",
+        "risetimedep=0.0",
+    ]
+    call(commands, stderr=PIPE)
+
+    ###
+    ### save STOCH
+    ###
+    stoch_file = simulation_structure.get_stoch_path(cs_root, name)
+    gen_stoch(stoch_file, srf_file)
+
+    ###
+    ### save INFO
+    ###
+    create_info_file(
+        srf_file,
+        1,
+        magnitude,
+        rake,
+        dt,
+        vs=vs,
+        rho=rho,
+        centroid_depth=depth,
+        lon=longitude,
+        lat=latitude,
+    )
+
+    # location of resulting SRF file
+    return srf_file
+
+
+def generate_sim_params_yaml(sim_params_file: str, parameters: Dict[str, Any]):
+    makedirs(path.dirname(sim_params_file), exist_ok=True)
+
+    sim_params = {}
+    for key, value in parameters.items():
+        if key in HF_RUN_PARAMS:
+            if "hf" not in sim_params.keys():
+                sim_params["hf"] = {}
+            sim_params["hf"][key] = value
+
+        elif key in BB_RUN_PARAMS:
+            if "bb" not in sim_params.keys():
+                sim_params["bb"] = {}
+            sim_params["bb"][key] = value
+
+        elif key in LF_RUN_PARAMS:
+            if "emod3d" not in sim_params.keys():
+                sim_params["emod3d"] = {}
+            sim_params["emod3d"][key] = value
+
+        else:
+            sim_params[key] = value
+
+    with open(sim_params_file, "w") as spf:
+        yaml.dump(sim_params, spf)
+
+
+def load_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("realisation_file", type=path.abspath)
+    parser.add_argument(
```

Comment:
as discussed this file should work on the 1, rather than having to need knowledge about our folder structure.

And then in slurm workflow a script to work on the many.

Reply:
That does sound harder in practice though.

Reply:
It does work on the one, it still needs to know where to put the Stoch, info and sim_params files though

Reply:
Sweet, so just needs a narrower path then.

Reply:
This script works on the 1. As it also creates the info/stoch/sim_params files it needs to know where to put those though.

Reply:
Silly git not updating that my message went through